### PR TITLE
upgrade spring and gradle versions (java 11 preparation)

### PR DESCRIPTION
**Pull request by ibilous**
Upgraded spring boot to 2.1.6 and gradle to 5.4.2. Gonna use these libs into java 11 project with appropriate spring and gradle versions. 
Conversion changes are not so big - dropwizard moved  few classes to separate packages/projects, the is new way how  need to connect lombok and KafkaBinderConfigurationProperties  removed `getConsumerConfiguration()` method, split execution  for several methods; the result we can get from `getConsumerProperties()`. I checked - looks code the same.

### CODE REVIEW

File: gradle/lib-versions.gradle [hunk 1]
```diff
@@ -13,8 +13,8 @@ ext.versions = [
     commonsCollections         : '3.2.2',
     commonsText                : '1.6',
 
-    springBoot                 : '2.0.6.RELEASE',
-    springCloud                : 'Finchley.RELEASE',
+    springBoot                 : '2.1.6.RELEASE',
+    springCloud                : 'Greenwich.RELEASE',
```
Comment (amedvedchuk) [inline gradle/lib-versions.gradle]: during previous migration to Spring Boot 2.0 we have used jhipster-dependencies 2.0.28 (https://mvnrepository.com/artifact/io.github.jhipster/jhipster-dependencies/2.0.28)

seems like we need to increase version to 3.0.x (https://mvnrepository.com/artifact/io.github.jhipster/jhipster-dependencies/3.0.4)

Comment (ibilous) [reply gradle/lib-versions.gradle]: just in order to clarify: i have made all according above comment

File: gradle/version.gradle [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import java.util.regex.Pattern
 
-version = "2.0.22"//detectSemVersion()
+version = "2.1.0"//detectSemVersion()
```
Comment (amedvedchuk) [inline gradle/version.gradle]: should we bump majpr version of commons to 3.0.0 ? as it is quite big changes.
@ishkurko @major13ua @sergeysenja1992  what do you think?

File: xm-commons-metric/src/main/java/com/icthh/xm/commons/metric/OperatingSystemGaugeSet.java [hunk 3]
```diff
@@ -26,72 +31,50 @@ public OperatingSystemGaugeSet(OperatingSystemMXBean mxBean) {
...
-        if (numeratorMethod.isPresent() && denominatorMethod.isPresent()) {
-            try {
-                long numerator = (long) numeratorMethod.get().invoke(operatingSystemMXBean);
-                long denominator = (long) denominatorMethod.get().invoke(operatingSystemMXBean);
-                if (0 == denominator) {
-                    return Double.NaN;
-                }
-                return 1.0 * numerator / denominator;
-            } catch (IllegalAccessException | InvocationTargetException ite) {
+    private double invokeRatio(String numeratorAttributeName, String denominatorAttributeName) {
```
Comment (amedvedchuk) [inline xm-commons-metric/src/main/java/com/icthh/xm/commons/metric/OperatingSystemGaugeSet.java]: try to rewrite in functional style.

File: gradle/version.gradle (hunk 2 above)
Comment (ishkurko) [reply gradle/version.gradle]: @amedvedchuk Propose to use 2.1.0 to align with spring boot.

Comment (amedvedchuk) [reply gradle/version.gradle]: @ishkurko we should not align to Spring boot version as it is separate component. I think we can not and should not put such constraint to xm-commons module

**Pull request merged**